File: app/src/main/java/pers/victor/androiddemos/activity/CallInterceptActivity.java
package pers.victor.androiddemos.activity;

import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;

import pers.victor.androiddemos.R;

/**
 * Created by Victor on 2017/4/1. (ง •̀_•́)ง
 */

public class CallInterceptActivity extends ToolbarActivity {
    public static String interceptNumber = "";

    @Override
    public int bindLayout() {
        return R.layout.activity_call_intercept;
    }

    @Override
    public void initView() {
        EditText editText = $(R.id.et_intercept_num);
        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                interceptNumber = s.toString();
            }
        });
    }
}

File: app/src/main/kotlin/pers/victor/androiddemos/activity/CallInterceptActivity.kt
package pers.victor.androiddemos.activity

import android.text.Editable
import android.text.TextWatcher
import pers.victor.androiddemos.R
import kotlinx.android.synthetic.main.activity_call_intercept.*

class CallInterceptActivity : ToolbarActivity() {
    companion object {
        var interceptNumber = ""
    }

    override fun bindLayout() = R.layout.activity_call_intercept

    override fun initView() {
        init()
    }

    private fun init() {

        et_intercept_num.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable) {
                interceptNumber = s.toString()
            }

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
            }

        })
    }

}
